lisa/transformers/kernel_installer.py

```diff
@@ -0,0 +1,308 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+
+import re
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Optional, Type, cast
+
+from dataclasses_json import CatchAll, Undefined, dataclass_json
+
+from lisa import schema
+from lisa.node import Node, RemoteNode
+from lisa.operating_system import Ubuntu
+from lisa.secret import PATTERN_HEADTAIL, add_secret
+from lisa.tools import Cat, Uname
+from lisa.transformer import Transformer
+from lisa.util import get_matched_str, subclasses
+from lisa.util.logger import Logger, get_logger
+
+
+@dataclass_json(undefined=Undefined.INCLUDE)
+@dataclass
+class BaseInstallerSchema(schema.TypedSchema):
+    delay_parsed: CatchAll = field(default_factory=dict)  # type: ignore
+
+
+@dataclass_json()
+@dataclass
+class RepoInstallerSchema(BaseInstallerSchema):
+    # version: str = "linux-image-azure-lts-18.04"
+    source: str = field(
+        default="proposed",
+        metadata=schema.metadata(
+            required=True,
+        ),
+    )
+    is_proposed: bool = True
```

Comment:
What is this about? Also the "proposed" string. Comments, at the very least?

Reply:
add comments for fields. the proposed field is used by Canonical for kernels before official release.